### PR TITLE
chore: Update to go 1.18 and other fixes

### PR DESCRIPTION
**Pull request by raulb**
## Description of change

- [chore: update man pages](https://github.com/meroxa/cli/commit/7b9aa3848ef3c734f144ac32c92d863d07f16655)
- [chore: update to go 1.18](https://github.com/meroxa/cli/commit/59f31370f3008d97912cef7c6050977673c19928)
- [feat: make production releases](https://github.com/meroxa/cli/commit/756fcf6f1be30d9b73f1a971d776794bb13c685d)
- [fix: unshallow error](https://github.com/meroxa/cli/commit/efa5e9a0a0f9585d2bda2e1387149efa22953d8c) 
- [fix: description on dispatcher updates](https://github.com/meroxa/cli/pull/362/commits/b1c362d51ad3930870c4568f978b3269cc5becfb)

## Type of change

- [ ]  New feature
- [x]  Bug fix
- [x]  Refactor
- [ ]  Documentation

### CODE REVIEW

**Comment by jayjayjpg** [comment]
@raulb Is it possible to test out the basic functionality of this PR first, by adding a condition to the workflow file, that allows the CI process to be triggered whenever a push is made to the `raul/fix-dispatch-event` feature branch?

This way, we may be able to catch anything that you may want to tweak before wrapping up, approving and merging the PR. What do you think?

File: .github/workflows/dispatch_turbine_go_dependency.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Remote Dispatch Turbine Go Dependency Responder
 
-on: [repository_dispatch]
+on: [workflow_dispatch,repository_dispatch]
```
Comment (jayjayjpg) [inline .github/workflows/dispatch_turbine_go_dependency.yml:3]: Could we do the following to trigger the workflow automatically on push? What do you think?

```suggestion
on: 
  workflow_dispatch
  repository_dispatch
  push:
    branches:    
      - 'raul/fix-dispatch-event'
```

Comment (raulb) [reply .github/workflows/dispatch_turbine_go_dependency.yml]: Leaving `workflow_dispatch` out for now. Testing on push

**Comment by raulb** [comment]
@jayjayjpg tested:

- https://github.com/meroxa/cli/runs/6576722975?check_suite_focus=true
- Dispatch one depends on turbine-go so I'll verify on the next update if that's ok

**Review by jayjayjpg — APPROVED** [review]
:sparkles: Awesome work! Thank you for putting this together :sparkles:

**Pull request merged**